sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala

```diff
@@ -88,9 +88,38 @@ case class Collate(child: Expression, collationName: String)
   override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode =
     defineCodeGen(ctx, ev, (in) => in)
 
-  override def sql: String = s"$prettyName(${child.sql}, $collationName)"
+  override def sql: String = s"$prettyName(${child.sql}, $collation)"
```

Comment:
for resolved collation, we should use the collation name instead of `ResolvedCollation(name)`

Reply:
I refactored the `ResolvedCollation.toString` method instead of modifying it in `Collate`.